lib/resque/job.rb

```diff
@@ -227,6 +256,20 @@ def args
       @payload['args']
     end
 
+    # Returns the metadata from the job's payload
+    def metadata
+      payload["meta"] ||= {}
+    end
```

Comment:
FEATURE: job metadata

This metadata is part of the job payload, separate from the arguments passed to the job's `perform` method.

This is used by the monolith for instrumentation.

This could be PRed to master, but will also need the accompanying change to `Resque::Job.destroy`.